tests/test_model_serializer.py

```diff
@@ -721,3 +721,59 @@ class Meta:
             str(exception),
             "Cannot set both 'fields' and 'exclude' options on serializer ExampleSerializer."
         )
+
+
+class Issue2704TestCase(TestCase):
+    def test_queryset_all(self):
+        class TestSerializer(serializers.ModelSerializer):
+            additional_attr = serializers.CharField()
+
+            class Meta:
+                model = OneFieldModel
+                fields = ('char_field', 'additional_attr')
+
+        OneFieldModel.objects.create(char_field='abc')
+        qs = OneFieldModel.objects.all()
+
+        for o in qs:
+            o.additional_attr = '123'
+
+        serializer = TestSerializer(instance=qs, many=True)
+
+        expected = [{
+            'char_field': 'abc',
+            'additional_attr': '123',
+        }]
+
+        assert serializer.data == expected
+
+
+class Issue2602TestCase(TestCase):
+    def test_manager_queryset_revaluation(self):
+        class TestSerializer(serializers.ModelSerializer):
+            class Meta:
+                model = OneFieldModel
+                fields = ('char_field',)
+
+        OneFieldModel.objects.create(char_field='123')
+
+        qs = OneFieldModel.objects
+        serializer = TestSerializer(instance=qs, many=True)
+
+        expected = [{
+            'char_field': '123'
+        }]
+
+        assert serializer.data == expected
+
+        for i in range(3):
+            OneFieldModel.objects.create(char_field=str(i))
```

Comment:
Yea I just noted that, my bad. Having a hard time trying to recreate whatever was going on in #2602